### ISSUE TITLE
add truncate=TRUE to glom() and argument names

### ISSUE THREAD

**Issue by dylanbeaudette** [ncss-tech/aqp#122]
Truncation at `z1` and `z2` thresholds, or horizon top/bottom if no data extending beyond those thresholds would support all manner of interesting analysis.

Also, can we consider re-naming the argument `as.data.frame` as it is rather common function? Perhaps `df` or something connotative.

A visual example so that I can understand the current implementation / results:
```r
library(aqp)

# fake data
set.seet(10101)
x <- lapply(1:10, random_profile, SPC=TRUE, method='LPP')
x <- union(x)

# glom-it!
y <- union(profileApply(x, glom, z1=25, z2=50))

# looks good
par(mar=c(0,0,3,1), mfrow=c(2,1))
plotSPC(x, color='p1', max.depth=120)
abline(h=c(25, 50), lty=2)

plotSPC(y, color='p1', max.depth=120)
abline(h=c(25, 50), lty=2)
```
![image](https://user-images.githubusercontent.com/624277/74281869-08a48980-4cd4-11ea-8ee7-7bc5a48ed37f.png)

**Comment by brownag** [comment]
```
# glom-it! truncate=TRUE works as expected
y2 <- union(profileApply(x, glom, z1=25, z2=50, truncate=TRUE))

# looks good
par(mar=c(0,0,3,1), mfrow=c(2,1))
plotSPC(x, color='p1', max.depth=120)
abline(h=c(25, 50), lty=2)

plotSPC(y2, color='p1', max.depth=120)
abline(h=c(25, 50), lty=2)
```

![image](https://user-images.githubusercontent.com/20842828/74282771-8cab4100-4cd5-11ea-9b03-a938892242b2.png)

**Comment by brownag** [comment]
added `ids` argument (short-circuit to get just hzIDs of intersection)
added `truncate` argument (clip upper and lower bounds  of result to z1 and z2)
renamed `as.data.frame` -> `df`

updated docs and tests

closed #122

**Comment by dylanbeaudette** [comment]
Thanks this is really useful.
